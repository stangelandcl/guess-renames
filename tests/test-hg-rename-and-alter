#!/bin/bash
# guess-renames
# Copyright (c) 2009 Colin Barrett <colin@springsandstruts.com>
#
# This software may be used and distributed according to the terms
# of the GNU General Public License, incorporated herein by reference.

export PYTHONPATH="$(dirname $(dirname $0))/testbase/lib/python:$PYTHONPATH"
export PATH="$(dirname $(dirname $0))/testbase/bin:$PATH"
echo "[extensions]" >> $HGRCPATH
echo "guessrenames.hgext=" >> $HGRCPATH

pushd $(dirname $(dirname $0)) > /dev/null
mkdir -p testbase/lib/python
python setup.py install --home=testbase > /dev/null
popd > /dev/null

mkdir testdir
pushd testdir > /dev/null

hg init
echo "*** Repository created."

(echo "alpha" && echo "bravo" && echo "charlie") > 1
(echo "delta" && echo "echo" && echo "foxtrot") > 2

echo "*** Files created. Contents:"

echo "*** 1:"
cat 1
echo "*** 2:"
cat 2

echo "*** Adding files."

hg st
hg addremove
hg st
hg commit -m "Adding"

echo "*** Files added."

echo "*** Renaming and altering."

rm -rf 1
rm -rf 2
(echo "one" && echo "bravo" && echo "charlie") > 3
(echo "four" && echo "echo" && echo "foxtrot") > 4

echo "*** Renamed and altered. Contents:"

echo "*** 3:"
cat 3
echo "*** 4:"
cat 4

echo "*** Guessing."

hg st
guess-renames
hg st
hg commit -m "Renaming"
hg st

echo "*** Done."

popd > /dev/null
rm -rf testdir

pushd $(dirname $(dirname $0)) > /dev/null
rm -rf testbase
popd > /dev/null

exit 0
