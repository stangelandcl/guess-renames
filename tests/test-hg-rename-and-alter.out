*** Repository created.
*** Files created. Contents:
*** 1:
alpha
bravo
charlie
*** 2:
delta
echo
foxtrot
*** Adding files.
? 1
? 2
adding 1
adding 2
A 1
A 2
*** Files added.
*** Renaming and altering.
*** Renamed and altered. Contents:
*** 3:
one
bravo
charlie
*** 4:
four
echo
foxtrot
*** Guessing.
! 1
! 2
? 3
? 4
2 is now 4 (score: 2.000000)
1 is now 3 (score: 2.000000)
A 3
A 4
R 1
R 2
*** Done.
